src/app/project/project.component.ts

```diff
@@ -13,82 +22,161 @@ import { PaginationInstance } from 'ngx-pagination';
+  constructor(
+    private projectService: ProjectService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private _changeDetectionRef: ChangeDetectorRef
+  ) {
+    this.objectFilterPipe = new ObjectFilterPipe();
+    this.proponentFilter = new ProponentFilterPipe();
+    this.projectTypeFilter = new ProjectTypeFilterPipe();
+    this.projectDecisionFilter = new ProjectDecisionFilterPipe();
+    this.filterPCP = new FilterPCPPipe();
+    this.projectPhaseFilter = new PhaseFilterPipe();
+  }
 
   ngOnInit() {
     this.loading = true;
-    this.projectService.getAll().subscribe(
-      data => {
-        this.results = data;
-        this.getProponents(data);
-        this.loading = false;
-        // Needed in development mode - not required in prod.
-        this._changeDetectionRef.detectChanges();
-      },
-      error => console.log(error)
-    );
+
+    this.route.params
+      .mergeMap((params: Params) => {
+        this.savedFilters = new ProjectFilters(params);
+        this.appliedFilters = new ProjectFilters(params);
+        this.showFilters = this.showAdvancedFilters();
+        return this.projectService.getAll();
+      })
+      .subscribe(
+        data => {
+          this.results = data;
+          this.distinctSortedProponentNames = this.getDistinctSortedProponentNames(this.results);
+          this.loading = false;
+          // Needed in development mode - not required in prod.
+          this._changeDetectionRef.detectChanges();
+        },
+        error => console.log(error)
+      );
   }
 
-  getProponents(projects) {
-    const names = [];
+  /**
+   * Returns an array of distinct, sorted proponent names.
+   * @param {Array<Project>} projects array of Projects.
+   * @returns {Array<string>} an array of distinct, sorted proponent names.
+   * @memberof ProjectComponent
+   */
+  getDistinctSortedProponentNames(projects: Array<Project>): Array<string> {
+    const names: Array<string> = [];
     projects.forEach(project => {
-      if (!project.proponent) {
-        // In case a proponent object isn't set yet.
-        project.proponent = { name: '' };
-      } else if (project.proponent.name && names.indexOf(project.proponent.name) === -1) {
-        // Keep unique list of names
+      if (project.proponent && project.proponent.name) {
         names.push(project.proponent.name);
-        // Add to proponent list.
-        this.proponents.push({ name: project.proponent.name });
       }
     });
+    return Array.from(new Set(names)).sort();
   }
 
-  sort (property) {
-    this.isDesc = !this.isDesc;
-    this.column = property;
-    this.direction = this.isDesc ? 1 : -1;
+  /**
+   * Sorts the page data by the specified column.
+   * @param {any} columnName the name of the column to sort by.
+   * @memberof ProjectComponent
+   */
+  sort(columnName: string) {
+    this.sortColumn = columnName;
+    this.sortDirection = this.sortDirection === 1 ? -1 : 1;
```

Comment:
Nit.: would something like `this.sortDirection = this.SortDirection * -1` work?

Reply:
Hmm yeah I guess so, will check and update